File: CostTrackerWebApp/web-app/src/app/shared/models/invoice.model.ts
import { BaseModel } from './base.model';
import { InvoiceMaterialModel } from './invoice-material.model';

export interface InvoiceModel extends BaseModel {
  description: string;
  invoiceDate: Date;
  supplierId: number;
  supplierName: string;
  totalPrice: number;
  materials: InvoiceMaterialModel;
}

File: CostTracker.Application/Models/InvoiceModel.cs
using System;
using System.Collections.Generic;

namespace CostTracker.Application.Models
{
    public class InvoiceModel
    {
        public int Id { get; set; }
        public string Description { get; set; }
        public DateTime InvoiceDate { get; set; }
        public int SupplierId { get; set; }
        public ICollection<InvoiceMaterialModel> Materials { get; set; }
    }
}
